src/shady-css/token-cursor.js

```diff
@@ -0,0 +1,36 @@
+const index = Symbol('index');
+const lexer = Symbol('lexer');
+const nextToken = Symbol('nextToken');
+
+class TokenCursor {
```

Comment:
I can eliminate the `offset` state from Tokenizer if it would make you feel better ;)